### DISCUSSION TITLE
Opening QuestPDF to other platforms. An API Server

### DISCUSSION THREAD

**Discussion by itsmuntadhar** [QuestPDF/QuestPDF#1035] [Show and tell]
We have been using QuestPDF and found it really great and easy with its fluent syntax, but, we wanted to make it available directly from the frontends and other languages so we created QuestPDF.Server an `ASP.NET Core` wrapper on top of it

We have open-sourced hoping someone else benefit from it.

You may find it [here](https://github.com/MorabaaSoftwareSolutions/QuestPDF.Server)

looking forward for your feedback

**Comment by MarcinZiabek** [comment]
Apologies for the late reply, and thank you for sharing this!

I really like your approach. Wrapping QuestPDF in an ASP.NET Core service is a clean way to make it accessible from other languages. Open-sourcing it so others can benefit is much appreciated.

Funnily enough, projects like yours were part of the inspiration a couple of years ago to think about bringing QuestPDF to other platforms. Now that .NET AOT compilation is stable and production-ready, we are actively exploring native ports to other languages - most likely Kotlin/Java and TypeScript first, with Python, PHP, and Go as possible follow-ups. The final scope will depend on developer experience, platform popularity, the effort required per port, and audience demand.

We already have a couple of internal prototypes working, so it's great to see that others have independently found value in using QuestPDF beyond the .NET ecosystem - it validates the direction.

Have you observed any usage of your project? Do you have any experience and know-how to share? 

Thanks again for sharing your work, and good luck with the project! ❤️
